### ISSUE TITLE
[Classic ERA] no addon detected

### ISSUE THREAD

**Issue by waytoomuchgaming** [ajour/ajour#649]
**Describe the bug**
Sometimes, Ajour don't find any addon (i have around 50 of them).
After some time, and multiple closing/reopening Ajour, it works again.

**Screenshots**
![Untitled0](https://user-images.githubusercontent.com/336778/120434408-be3d9500-c37c-11eb-882d-b745aa5eb334.png)
![Untitled](https://user-images.githubusercontent.com/336778/120434012-348dc780-c37c-11eb-8d8f-7aa6227f2dbe.png)

**Software involved**
 - OS: Windows 10
 - Ajour version: 1.2.0
 - Classic ERA

**Comment by waytoomuchgaming** [comment]
As it take some time to "fail", i guess addons not found are not the ones on my computer, but the ones online.

**Comment by casperstorm** [comment]
It sounds like an upstream issue with one of the API's. They seem to respond quite slow so maybe one were down. If it happens again, can you try to locate the ajour.log file (Settings -> Open data Directory) and paste it here?
Then i'll take a look to confirm where the problem is.

**Comment by Gillingham** [comment]
Encountering the same issue on macOS. Is there a way to turn on a higher level debug, it's not clear to be what request is failing:
```
08:24:17.344 [ajour::gui][DEBUG] antialiasing: true
08:24:17.425 [ajour_core::utility][DEBUG] checking for application update
08:24:17.425 [ajour_core::theme][DEBUG] loading user themes
08:24:17.426 [ajour_core::fs::theme][DEBUG] loaded 0 user themes
08:24:17.517 [ajour::gui::update][DEBUG] Message::CachesLoaded(error: false)
08:24:17.517 [ajour::gui::update][DEBUG] Message::ThemesLoaded(0 themes)
08:24:17.528 [ajour::gui::update][DEBUG] Message::Parse
08:24:18.143 [ajour_core::catalog][DEBUG] Downloaded latest catalog from https://github.com/ajour/ajour-catalog/raw/master/catalog-3.0.json
08:24:18.160 [ajour::gui::update][DEBUG] Message::LatestRelease(Some("1.2.0"))
08:24:18.492 [ajour::gui::update][DEBUG] Message::CatalogDownloaded(16473 addons in catalog)
08:24:19.215 [ajour::gui::update][DEBUG] Interaction::SelectWowDirectory(None)
08:24:25.953 [ajour::gui::update][DEBUG] Message::UpdateWowDirectory(Chosen(Some("/Applications/World of Warcraft/_classic_") - None))
08:24:25.953 [ajour::gui::update][DEBUG] Message::UpdateWowDirectory(Resolution("/Applications/World of Warcraft"))
08:24:25.962 [ajour::gui::update][DEBUG] Message::Parse
08:24:25.963 [ajour::gui::update][DEBUG] preparing to parse addons in "/Applications/World of Warcraft/_classic_/Interface/AddOns"
08:24:25.963 [ajour_core::parse][DEBUG] Classic TBC - parsing addons folder
08:24:25.963 [ajour_core::parse][DEBUG] Classic TBC - 43 folders in AddOns directory to parse
08:24:25.964 [ajour::gui::update][DEBUG] Message::CheckWeakAurasInstalled(Classic TBC, is_installed: false)
08:24:26.225 [ajour_core::parse][DEBUG] Classic TBC - 43 fingerprints: 0 cached, 43 calculated, 43 added, 0 removed
08:24:26.228 [ajour_core::parse][DEBUG] Classic TBC - 43 addon folders successfully parsed from '.toc'
08:24:26.228 [ajour_core::parse][DEBUG] Classic TBC - 0 valid cache entries retrieved
08:24:26.228 [ajour_core::parse][DEBUG] Classic TBC - 43 unique fingerprints to check against curse api
08:24:26.320 [ajour][ERROR] Failed to parse addons
08:24:26.320 [ajour][ERROR] caused by: failed to connect to the server
08:24:26.320 [ajour][ERROR] caused by: [35] SSL connect error
```

**Comment by casperstorm** [comment]
This is actually enough information for me:

```
08:24:26.320 [ajour][ERROR] Failed to parse addons
08:24:26.320 [ajour][ERROR] caused by: failed to connect to the server
08:24:26.320 [ajour][ERROR] caused by: [35] SSL connect error
```

It failed to get a connection with the API, most likely Curse. So it is one of two things:

1. Curse API was down.
2. Your Internet is blocking access.

If it keeps happening, it sounds like a issue on your end with their server. Like, [Pi-hole](https://pi-hole.net/) eg.

**Comment by waytoomuchgaming** [comment]
Maybe you can change the error message for end users ?
Because "you have no classic era addons" can be a bit disturbing to read ^^

**Comment by casperstorm** [comment]
> Maybe you can change the error message for end users ?
> 
> Because "you have no classic era addons" can be a bit disturbing to read ^^

Yes!
It's ... pretty bad UX. Ill try to find a better solution.

**Comment by casperstorm** [comment]
I have begun working on the UX part of this now.

**Comment by casperstorm** [comment]
Released in [1.2.3-rc.1](https://github.com/ajour/ajour/releases/tag/1.2.3-rc.1).
